### PR TITLE
Unit 4.1 Component User List Array

### PR DESCRIPTION
**Pull request by nathanjophillips**
## The Purpose
- To learn how to work with nested components and binding data with Polymer
- To generate a working list of users to use in further testing throughout the build of this project

## What It Does
1. This imports `user-profile` into `user-list` to create simplicity for the project, so each component has a very specific purpose

2. The properties inside `user-profile` are one way binded to be accessed for use in the created array in the `user-list`

`[[id]] [[firstName]] [[lastName]] [[phoneNumber]] [[email]]`

3. Then the properties are assigned in the `user-list` like this with a `dom-repeat`: 
 
Screen Shot 2019-05-09 at 11 28 49 AM

4. The `user-list` component is then imported into `intern-app` while adding dummy data into the defined array to display the given info on each user profile card 

5. Additionally a `dom-if` was added to the `user-profile` component to create a new function of `isExpanded` to keep the expand/collapse button working on each card 

## This accomplishes the following requirements given for  [Unit 4.1](https://github.com/phillnab/nathan-phillips-ux-iop/blob/master/issues/unit-4/4.1-polymer-components.md)
- The following HTML pages are represented in the Polymer app:
    - [x] user-list.html
    - [x] user-profile.html
    - [x] create-user.html
    - [x] edit-user.html
- [x] The `user-list` element has `user-profile` components nested inside it
- [x] The `user-profile` component maintains its expand and collapse functionality

### The changes to look for are:
1. Check if the expand/collapse functionality work 
2. Check if each card is displaying with different content
3. Any syntax errors

# How To Test
- Clone Repo [nathan-phillips-ux-iop](https://github.com/phillnab/nathan-phillips-ux-iop)
- Checkout branch `nathan-phillips-ux-iop/unit-4.1-polymer-component-user-list`
- Run `yarn`
- Run `yarn build`
- Run `yarn start`

### See if it matches this:

Screen Shot 2019-05-09 at 12 56 22 PM

## Checklist

Under penalty of public shaming, I avow that I:

- [x] Reviewed the diff to look for typos, whitespace errors, and console.log() statements
- [x] Verified that there are no compiler or linter errors or warnings
- [ ] Verified the build in production(optimized) mode
- [x] Updated the docs, if necessary
- [x] Included the appropriate labels
- [x] Referenced applicable issues (i.e. Closes #XXXX, Fixes #XXXX)

Browsers tested:
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] Internet Explorer

### CODE REVIEW

File: src/components/intern-app.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <link rel="import" href="~@banno/polymer/polymer-element.html">
-<link rel="import" href="./user-profile.html">
+<link rel="import" href="~@banno/polymer/lib/elements/dom-repeat.html">
```
Comment (kddrake) [inline src/components/intern-app.html]: don't need this import here. only need to import it if you're using it in the file in which you're importing.

File: src/components/intern-app.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 <!-- <link rel="import" href="./create-user.html"> -->
```
Comment (kddrake) [inline src/components/intern-app.html]: let's get rid of these commented out imports

File: src/components/intern-app.html [hunk 3]
```diff
@@ -20,33 +20,33 @@
...
-                lastName: "Cage",
-                phoneNumber: 8675309,
-                email: "nick@nickcage.com"
-              }
-            },
-          },
-        };
+          userProfiles: {
+            type: Array,
+            value() {
```
Comment (kddrake) [inline src/components/intern-app.html]: needs to be
```suggestion
            value: () => {
```

File: src/components/user-profile.html [hunk 4]
```diff
@@ -62,18 +61,18 @@ <h3>User Profile Component</h3>
...
+      }
+      buttonText(isExpanded) {
         return isExpanded ? `Collapse` : `Expand`;
       }
     }
     customElements.define(UserProfile.is, UserProfile);
     export default UserProfile;
   </script>
-</dom-module>
+    </dom-module>
```
Comment (kddrake) [inline src/components/user-profile.html]: remove the whitespace before the tag

**Review by kddrake — CHANGES_REQUESTED** [review]

**Review by Jropp — APPROVED** [review]
Other than the merge conflict it looks good and functions as expected for me

**Review by kddrake — APPROVED** [review]

**Pull request merged**